app/controllers/api/connections_controller.rb

```diff
@@ -55,4 +64,104 @@ def request_match
 
     end
 
+    def update_match
+
+        if @current_user
+
+            request = MatchRequest.find_by(id: params[:id])
+
+            if request
+                #check if request exists
+
+                if request.status_is_pending?
+                    
+                    begin
+                        request.status = params[:status]
+                    rescue
+                        error = {"error": "Updated status is invalid"}
+                        render(json: error, status: 400)
+                        return
+                    end
```

Comment:
Have you checked if the `rescue` block is ever used? `request.status = params[:status]` shouldn't raise an error. If you really want to make sure that the status is valid, you can keep line 79, but instead of `begin` and `rescue` just do `unless request.valid?`

Reply:
Yes, I put the rescue block specifically because it raised an exception, since status is an enum.